src/BayesFilters/src/EstimatesExtraction.cpp

```diff
@@ -235,11 +236,20 @@ VectorXd EstimatesExtraction::map
      * Saha, S., Boers, Y., Driessen, H., Mandal, P. K., Bagchi, A. (2009),
      * 'Particle Based MAP State Estimation: A Comparison.',
      * 12th International Conference on Information Fusion,
-     * Seattle, WA, USA, July 6-9, 2009. */
+     * Seattle, WA, USA, July 6-9, 2009. 
+     *
+     * The equation is rewritten in order to work in the log space.
+     */
 
     ArrayXd::Index map_index;
 
-    ((transition_probabilities * previous_weights).array() * likelihoods.array()).maxCoeff(&map_index);
+    VectorXd values(particles.cols());
+
+    double eps = std::numeric_limits<double>::min();
+    for (std::size_t i = 0; i < values.size(); i++)
+        values(i) = log(likelihoods(i) + eps) + utils::log_sum_exp((transition_probabilities.row(i).array() + eps).log() + previous_weights.array());
```

Comment:
I don't really like to pass eps to logsumexp as the only point of failure would be the log evaluated in 0.
To be precise, a good apporach would be to have inside logsumexp an eps summed only in the evaluation of the log.
Anyway, let's leave it like this and keep in mind that we may change it in some future commits.

Reply:
You are right, I agree on changing it.